### ISSUE TITLE
.NET on WebAssembly - Bindings to Sentry's JavaScript SDK

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry-dotnet#3673]
The `Sentry` package as well as the new `Sentry.AspNetCore.Blazor.WebAssembly` work on the browser through wasm.

Related work:
* https://github.com/getsentry/sentry-dotnet/pull/2569
* https://github.com/getsentry/sentry-dotnet/issues/317
* https://github.com/getsentry/sentry-dotnet/pull/3501

We have a sample in the repo:
* https://github.com/getsentry/sentry-dotnet/pull/574

But none of the work done yet (other than the [draft PR from hackweek](https://github.com/getsentry/sentry-dotnet/pull/2569) have done anything related to bundling our JS SDK. 
This should allow us to get [Session Replay](https://docs.sentry.io/product/explore/session-replay/web/), our [User Feedback Widget](https://docs.sentry.io/product/user-feedback/#user-feedback-widget) as well as some JS specific features like Unhandled Promise Rejections, etc.
 
Related issues and PRs:
* https://github.com/dotnet/runtime/issues/91049
* https://github.com/getsentry/sentry-unity/issues/668
* https://github.com/getsentry/sentry-godot/issues/114
* https://github.com/getsentry/sentry-dart/pull/2238/
* https://github.com/getsentry/sentry-kotlin-multiplatform/issues/27
* https://github.com/getsentry/sentry-javascript/issues/8259#issuecomment-1570196653
* https://github.com/getsentry/sentry-javascript/pull/8263
* https://github.com/getsentry/sentry-dotnet/issues/2291

Package [is still `-preview`](https://github.com/getsentry/sentry-dotnet/pull/3674), but got some downloads already
Tracking adoption of Blazor specific wasm package: https://nugettrends.com/packages?months=6&ids=Sentry.AspNetCore.Blazor.WebAssembly

![Image](https://github.com/user-attachments/assets/91d3a3a4-e057-48fc-bbe6-3ac256885c5e)

**Comment by bruno-garcia** [comment]
Symbolication is blocked by:
* https://github.com/dotnet/runtime/issues/96334

**Comment by MaxwellDAssistek** [comment]
This would also be nice to have for Server Interactive Blazor mode.

**Comment by bruno-garcia** [comment]
> This would also be nice to have for Server Interactive Blazor mode.

Are those errors not captured today? if it's a C# error it should already be captured regardless if thrown client or server side. The JS SDK would help capture JS errors that don't get into the C# handler (P/Invoke should).

**Comment by cobyeastwood183** [comment]
https://sentry.zendesk.com/agent/tickets/147928

**Comment by klemmchr** [comment]
I would love to see this implemented to get Session Replay in my Blazor WASM application. Is there any estimation on when this could be used?

**Comment by jamescrosswell** [comment]
@klemmchr no time frames for implementation yet no.
